Gems/Atom/RPI/Code/Source/RPI.Reflect/Material/MaterialAsset.cpp

```diff
@@ -182,6 +194,52 @@ namespace AZ
             m_isDirty = false;
         } 
 
+        template <typename iteratorType>
+        void MaterialAsset::RenamePropertyNames(iteratorType start, iteratorType end)
+        {
+            for (iteratorType it = start; it != end; ++it)
+            {
+                for (auto& propertyName : m_propertyNames)
+                {
+                    const auto toFromIterator = it->find(propertyName.GetCStr());
+                    if (toFromIterator != it->end())
+                    {
+                        propertyName = AZ::Name{ toFromIterator->second };
+                    }
+                }
+            }
+        }
+
+        void MaterialAsset::RenamePropertyNames()
+        {
+            // construct toFrom map in ascending order of version.
+            AZStd::vector<AZStd::map<AZStd::string, AZStd::string>> versionToFrom;
+
+            const bool ascending = m_materialTypeVersion < m_materialTypeAsset->GetVersion();
```

Comment:
Required: only support ascending version numbers. We should not commit to supporting downgrading.
It will also simplify the below code.